src/Workspaces/Core/Portable/Workspace/Solution/ProjectDependencyGraph_AddProjectReference.cs

```diff
@@ -34,7 +34,7 @@ internal ProjectDependencyGraph WithAdditionalProjectReferences(ProjectId projec
             // Note: rather than updating our dependency sets and topologically sorted data, we'll throw that away since incremental update is
             // tricky, and those are rarely used. If somebody needs them, it'll be lazily computed.
             return new ProjectDependencyGraph(
-                _projectIds,
+                _projectIds.Union(referencedProjectIds),
```

Comment:
See comment elsewhere: the existing code was correct (or at least matched the other code); if you're seeing a failure here it means some higher code is failing to filter it out.